scripts/zkai_native_attention_mlp_single_proof_route_gate.py

```
@@ -0,0 +1,632 @@
+        "validation_commands": list(VALIDATION_COMMANDS),
+        "route_commitment": route_commitment(routes),
+    }
+    refresh_payload_commitment(payload)
+    return payload
+
+
+def validate_payload(payload: Any, *, expected: dict[str, Any] | None = None, context: dict[str, Any] | None = None) -> None:
+    data = _dict(payload, "payload")
+    key_set = set(data)
+    if key_set not in (CORE_KEYS, FINAL_KEYS):
+        raise NativeAttentionMlpSingleProofRouteError(f"unexpected payload keys: {sorted(key_set ^ FINAL_KEYS)}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)